exe/Parser/Text.hs

```diff
@@ -527,33 +527,50 @@ moduleToFileName = map (\x -> if x == '.' then '/' else x)
 dirFromFileName :: String -> String
 dirFromFileName = reverse . dropWhile (/= '/') . reverse
 
-generateModule ::
+type GeneratorRecipe a = (String, String -> Fold IO a String)
+
+fileEmitter :: String -> GeneratorRecipe a -> Fold IO a ()
+fileEmitter outdir (modName, fldGen) = Fold.mapM action $ fldGen modName
+
+    where
+
+    outfile = outdir <> moduleToFileName modName <> ".hs"
+    outfiledir = dirFromFileName outfile
+    action c = createDirectoryIfMissing True outfiledir >> writeFile outfile c
+
+generateOneModule ::
        String
     -> (SerialT IO String -> SerialT IO a)
     -> String
+    -> GeneratorRecipe a
+    -> IO ()
+generateOneModule infile transformLines outdir recipe =
+    readLinesFromFile infile & transformLines
+        & Stream.fold (fileEmitter outdir recipe)
+
+generateMultipleModules ::
+       String
+    -> (SerialT IO String -> SerialT IO a)
     -> String
-    -> (String -> Fold IO a String)
+    -> [GeneratorRecipe a]
     -> IO ()
-generateModule infile transformLines outdir modName fldGen =
-    readLinesFromFile infile & transformLines & Stream.fold moduleFold
+generateMultipleModules infile transformLines outdir recipes =
+    readLinesFromFile infile & transformLines & Stream.fold combinedFld
```

Comment:
Instead of `generateMultipleModules` we can write a `combineRecipes` function to combine multiple recipes into one and then use `generateOneModule` and `combineRecipes` at the call site.